stable/grafana/templates/_helpers.tpl

```diff
@@ -30,3 +30,71 @@ Create chart name and version as used by the chart label.
 {{- define "grafana.chart" -}}
 {{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{/*
+Create the LDAP configuration for Grafana
+*/}}
+{{- define "ldap.toml" -}}
+{{- $ldapToml := index .Values "ldap.toml" }}
```

Comment:
Why don't you put this into a configmap whose content a user can fully control themselves?

Reply:
@unguiculus I think the reason is that this PR builds a Secret, that uses this template to populate it, based on an entry on values.yaml so it maps toml to yaml on this template in order to make easier to write the entries in values.yaml.

Although this gives flexibility I agree that maybe in this case is better to simply populate ldap.toml from a Secret (I prefer over ConfigMap) configured by the user or if not provide default values with plain .toml on values.yaml, if not maintaining all possible options for this config file might be more a headache than a long term solution.

Reply:
I'd really go for the secret approach.

Reply:
@unguiculus  For maximum flexibility would we want the user to be able to define the content of their `ldap.toml` in a vaule in then convert that into a secret. I originally chose this approach as I was going to use `toToml` with `ldap.toml` to generate the TOML and allow for the users to be able to define their configuration as an object, but integers being converted to floats prevented that. 

e.g. 

```yaml
ldap.toml: |
[[servers]]
host: ...
...
```

Reply:
Why not let the user decide if they want to use their own Secret or let the Chart create one.
For example on values.yaml:
```yaml
ldap:
  secret:
    create: true
   ## Default {{ template "grafana.fullname" . }} or overwritten by this value
    name: ""
  ## ldap.toml definition or using yaml to toml mapping
  config: {}
```
This example will create the secret and populate it using `ldap.config`

If the user wants to use its own Secret
```yaml
ldap:
  secret:
    create: false
    name: "my-grafana-ldap-secret"
  config: {}
```
`config:` is irrelevant in this case and an external secret will be used.

Reply:
Updated to allow for an existing secret or allow for the creation of one from the chart